src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -87,6 +92,45 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         }
     }
 
+    private bool ShouldDropMessage(CmsEvent cmsEvent)
+    {
+        if (cmsEvent == CmsEvent.CREATE)
+        {
+            Logger.LogInformation("Dropping received event {cmsEvent}", cmsEvent);
+            return true;
+        }
+
+        if (ShouldIgnoreMessage(cmsEvent))
+        {
+            Logger.LogInformation("Receieved {event} but UsePreview is {usePreview} - dropping message", cmsEvent, _contentfulOptions.UsePreview);
+            return true;
+        }
+
+        return false;
+    }
+
+    private bool IsNewAndValidComponent(ContentComponentDbEntity mapped, ContentComponentDbEntity? existing)
+    {
+        if (existing != null) return true;
+
+        if (AnyRequiredPropertyIsNull(mapped))
+        {
+            Logger.LogInformation("A required property of the component with ID {id} is null, so the message will be dropped!", mapped.Id);
```

Comment:
Probably worth changing `AnyRequiredPropertyIsNull` to return the erroneously null property names instead, and then logging those instead. Give us better insight incase something else goes skewif

Reply:
Resolved in: [b264fe0](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/508/commits/b264fe05eae860b4c7949b9859b54600e4e6f124)